proposals/lip-0044.md

```diff
@@ -596,29 +646,71 @@ def getGeneratorsBetweenTimestamps(startTimestamp: uint32, endTimestamp: uint32)
 
 #### setGeneratorList
 
-This function updates the value of the generator list to the input `generatorAddresses`. It checks if all elements of `generatorAddresses` have a registered validator account in the validators data substore. If this is not the case, it throws an error. Otherwise, it updates the value of `generatorList`.
+#### setValidatorParams
+
+This function allows to set the thresholds, validators and associated BFT weights to be used from the next height onward and updates the BFT module store accordingly.
+
+##### Parameters
+
+* `precommitThreshold`: The precommit threshold value to be used from the next height onward. The value must be a 64-bit unsigned integer.
+* `certificateThreshold`: The certificate threshold value to be used from the next height onward. The value must be a 64-bit unsigned integer.
+* `validatorList`: The validators and their associated BFT weight to be used from the next height onward. The value must be an array of objects with an `address` property containing the 20-byte address of the validator and a `bftWeight` property containing the BFT weight of the validator as 64-bit unsigned integer. Note that this function fails if the number of provided validators is more than `LSK_BFT_BATCH_SIZE`.
+
+##### Execution
 
 ```python
-def setGeneratorList(generatorAddresses: list[Address]) -> None:
-    for address in generatorAddresses:
-        if no entry in the validators data substore exist with storeKey == address:
-            raise Exception(f'Invalid address in generator list. The address {address.hex()} is not registered as validator.')
-    generatorList = generatorAddresses
+setValidatorParams(precommitThreshold: uint64, certificateThreshold: uint64, validatorList: list[object]):
```

Comment:
I tried to put the function from the BFT module here 1-1 and also store exactly the same information as in the BFT Params store, but of course we could change it. I was thinking it would be nice to do the validation here so if in the application layer this method is called with invalid parameters, an exception is raised directly. Additionally, I though it is nice to have all consensus-related properties here in one place as they can then be also requested via an endpoint. However, in terms of protocol it would not be necessary to store them. I am open to suggestions here.

Reply:
I think it would be nicer if we can avoid validators module to know too much about BFT protocol, so I would prefer to keep it as proxy to reply to the engine which stores the value as it is.

Validation wise it will be the same in most cases. Even if we don't throw in the application, it will be thrown in the engine and the block will be invalid.
Maybe only case where it could be different is if there is integration test without engine, developer will not notice, but I think that's probably true for other replies for engine (like invalid event schema etc), also this can be resolved by mock ABI implementation.

Currently, there is `consensus` endpoint exposed from the engine, so BFT related properties can be fetched from the endpoint (for certificate etc)

Reply:
Okay, then in terms of specifications, the function `setValidatorParams` is a simple getter without validation, but in the BFT LIP I keep the full function with full validation. I will also remove the `prevoteThreshold` and `validatorsHash` property from the store.

Reply:
This comment can be resolved from my side.